.github/workflows/deploy.yaml

```diff
@@ -21,7 +21,24 @@ jobs:
         with:
           persist-credentials: false
 
-      - uses: JamesIves/github-pages-deploy-action@4.1.4
+      - name: Read .nvmrc
```

Comment:
Rather than repeating the build and test steps in separate action files, can we factor them out to a composite run steps action? https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action
This will prevent us from making changes in one place but not the other in the future.